### DISCUSSION TITLE
3.6.0 - unflattening configuration doesn't seem to work

### DISCUSSION THREAD

**Discussion by IanKemp** [riok/mapperly#1485] [Q&A]
I have some classes and interfaces:

```csharp
interface IExternalRequest
{
    int? UserId { get; set; }
    string? UserName { get; set; }
}

record SomeExternalRequest(
    int Foo,
    bool Bar,
    int? UserId,
    string? UserName)
    : IExternalRequest;
```

I want to unflatten these into:

```csharp
record InternalUser(
    int? Id,
    string? Name);

record SomeInternalRequest(
    int Foo,
    bool Bar,
    InternalUser User);
```

So I created a user-defined mapping for `IExternalRequest` => `InternalUser`:

```csharp
internal static class IExternalRequestMapper
{
    public static InternalUser MapToUser(this IExternalRequest request)
        => new(
            request.UserId,
            request.UserName);
}
```

and then another for `SomeExternalRequest` => `SomeInternalRequest`:

```csharp
[Mapper]
[UseStaticMapper(typeof(IExternalRequestMapper))]
internal static partial class SomeRequestMapper
{
    public static partial SomeInternalRequest Map(this SomeExternalRequest request);
}
```

Unfortunately this doesn't work - Mapperly complains that "SomeInternalRequest has no accessible constructor with mappable arguments", as well as that `UserId` and `UserName` are not mapped from the source and `User` is not found on the source.

I then tried adding `MapPropertyFromSourceAttribute`:

```csharp
[Mapper]
[UseStaticMapper(typeof(IExternalRequestMapper))]
internal static partial class SomeRequestMapper
{
    [MapPropertyFromSource(nameof(SomeInternalRequest.User), Use = nameof(IExternalRequestMapper.MapToUser))]
    public static partial SomeInternalRequest Map(this SomeExternalRequest request);
}
```

but that complains that "The referenced mapping named MapToUser was not found".

Am I doing something wrong, or is this not a supported scenario by Mapperly?

**Comment by latonz** [comment]
The usage of external mappers in `MapPropertyFromSourceAttribute.Use` is currently not supported, see https://github.com/riok/mapperly/issues/1298
